core/src/query.rs

```diff
@@ -59,90 +60,118 @@ impl MongoQuery {
+                doc! {"$project":{
+                    "_id": 1,
+                    "schema": 1
                     }
-                }
-            },
-            doc! {"$project": {
-                "_id": 0,
-                current_db: {
-                    "$arrayToObject": [{
-                        "$map": {
-                            "input": "$collections",
-                            "as": "coll",
-                            "in": {
-                                "k": "$$coll.collectionName",
-                                "v": "$$coll.schema"
-                                }
+                },
+                doc! {"$group": {
+                    "_id": null,
+                    "collections": {
+                        "$push": {
+                            "collectionName": "$_id",
+                            "schema": "$schema"
                             }
-                        }]
+                        }
                     }
-                }
-            },
-        ];
+                },
+                doc! {"$project": {
+                    "_id": 0,
+                    current_db: {
+                        "$arrayToObject": [{
+                            "$map": {
+                                "input": "$collections",
+                                "as": "coll",
+                                "in": {
+                                    "k": "$$coll.collectionName",
+                                    "v": "$$coll.schema"
+                                    }
+                                }
+                            }]
+                        }
+                    }
+                },
+            ];
+
+            // create the schema_catalog document
+            let schema_catalog_doc_vec: Vec<Document> = client.runtime.block_on(async {
+                schema_collection
+                    .aggregate(schema_catalog_aggregation_pipeline)
+                    .await
+                    .map_err(Error::QueryExecutionFailed)?
+                    .try_collect::<Vec<Document>>()
+                    .await
+                    .map_err(Error::QueryExecutionFailed)
+            })?;
 
-        // create the schema_catalog document
-        let schema_catalog_doc_vec: Vec<Document> = client.runtime.block_on(async {
-            schema_collection
-                .aggregate(schema_catalog_aggregation_pipeline)
-                .await
-                .map_err(Error::QueryExecutionFailed)?
-                .try_collect::<Vec<Document>>()
-                .await
-                .map_err(Error::QueryExecutionFailed)
-        })?;
+            if schema_catalog_doc_vec.len() > 1 {
+                return Err(Error::MultipleSchemaDocumentsReturned(
+                    schema_catalog_doc_vec.len(),
+                ));
+            } else if schema_catalog_doc_vec.is_empty() {
+                return Err(Error::SchemaCollExistsButNoSchemaInformationWasReturned(
```

Comment:
I am not sure this case is ever hit. And if it is, shouldn't we assign empty schemas to all the collections in the namespace?

Reply:
This case is hit if you doing something like `SELECT * FROM <nonexistent_collection>`. Basically, the aggregation pipeline returns nothing, so then the `schema_catalog_doc_vec` is empty. This happens if you have a `__sql_schemas` collection in the DB and namespaces in the query, but none of the namespaces have an entry in the `__sql_schemas` collection. At first, I thought this was such a bizarre case that we should return an error, but I saw what you did the JDBC driver if this case is hit, and I think it's a better approach (i.e, put a warning in the logs and generate empty schemas). I'll modify the code to do that instead 👍.